src/client.js

```diff
@@ -1,25 +1,11 @@
-let isNode = false;
-
 const VERSION = '0.0.3';
 const RETRY_STATUS_CODES = [429, 500, 502, 503, 504];
 const ENDPOINT = 'https://api.mistral.ai';
 
-/**
- * Initialize fetch
- * @return {Promise<void>}
- */
-async function initializeFetch() {
-  if (typeof window === 'undefined' ||
-    typeof globalThis.fetch === 'undefined') {
-    const nodeFetch = await import('node-fetch');
-    fetch = nodeFetch.default;
-    isNode = true;
-  } else {
-    fetch = globalThis.fetch;
-  }
-}
-
-initializeFetch();
+// We can't use a top level await if eventually this is to be converted
+// to typescript and compiled to commonjs, or similarly using babel.
+const configuredFetch = Promise.resolve(
+  globalThis.fetch ?? import('node-fetch').then((m) => m.default));
```

Comment:
Some credit to [fuegoio](https://github.com/fuegoio) here:
https://github.com/mistralai/client-js/pull/65#pullrequestreview-2028939725